### ISSUE TITLE
⌨ Notifications: Screen reader users are not informed of outstanding notifications

### ISSUE THREAD

**Issue by MaddyDaigle** [GCTC-NTGC/gc-digital-talent#12891]
## ⌨ Accessibility Issue

When a screen reader user navigates to the notification button on the application, they are not informed of outstanding notifications. Visual users can see a small dot indicating a new notification, but this information is not conveyed to screen reader users, creating an unequal experience. Additionally, the dot could be made more prominent for sighted users.

### WCAG 2.1 Criteria

WCAG 1.3.1 Info and Relationships – Information, structure, and relationships conveyed visually must also be programmatically determined or available in text.

WCAG 4.1.2 Name, Role, Value – User interface components must provide meaningful information to assistive technologies.

## 🦋 Expected Behaviour

When a screen reader user navigates to the notification button, they should be informed of outstanding notifications (e.g., "Notifications (1 unread)"). This ensures equal access to information. Additionally, the visual indicator could be enhanced to make it more noticeable to sighted users.

## 🕵️ Details

- **Operating System**: MacOS
- **Browser**: Chrome
- **Assistive Technology**: Voiceover

## 📋 Steps to Reproduce

Tab to notification button from anywhere on the application.

## 📸 Screenshot

![Image](https://github.com/user-attachments/assets/e4f18ea7-61a7-47ee-9bd2-2db242724389)

## 🙋‍♀️ Proposed Solution

Update the `aria-label` to include a count of unread notifications when they exist.

```tsx
aria-label={intl.formatMessage({
  defaultMessage: "View notifications{count}",
  id: "ztx8xL",
  description: "Button text to open the notifications dialog",
}, {
  count: notificationCount > 0 ? intl.formatMessage({ 
    defaultMessage: " ({ count} unread)", { count: notificationCount }
  })  : ""
})}
```

## ✅ Acceptance Criteria

- [ ] Notification dialog indicates number of unread notifications to screen readers
